src/zappy_gui_src/Core/SFML/Interface/Inventory/Inventory.cpp

```diff
@@ -5,35 +5,149 @@
 ** Inventory
 */
 
+/// \file src/zappy_gui_src/Core/SFML/Interface/Inventory/Inventory.cpp
+
 #include "Inventory.hpp"
+#include "ZappyGuiException.hpp"
+#include <string>
+#include <iostream>
 
 using namespace gui;
 
 Inventory::Inventory()
 {
+    std::vector<int> tmpvec = {-1, -1, -1, -1, -1, -1, -1};
+    _tileInventory = tmpvec;
 }
 
 Inventory::Inventory(std::shared_ptr<sf::RenderWindow> window)
 {
     setWindow(window);
 }
 
-void Inventory::initialize_shapes()
+void Inventory::initShapes()
 {
     sf::Vector2u windowSize = _window->getSize();
 
+    _isOpen = true;
+    _body.setOutlineThickness(1);
     _body.setSize(sf::Vector2f(windowSize.x, windowSize.y / 5));
     _body.setPosition(sf::Vector2f(0, windowSize.y / 5 * 4));
-    _body.setFillColor(sf::Color::Cyan);
     _button.setPosition(sf::Vector2f(_body.getPosition().x + 10, _body.getPosition().y + 10));
-    _button.setSize(sf::Vector2f(10, 10));
-    _button.setFillColor(sf::Color::Magenta);
+    _button.setPointCount(3);
+    _button.setPoint(0, {0, 10});
+    _button.setPoint(1, {5, 0});
+    _button.setPoint(2, {10, 10});
+    if (!_font.loadFromFile(FONT_PATH))
+        throw InventoryException("Inventory Exception", "Could not load font");
+    _title.setFont(_font);
+    _title.setCharacterSize(20);
+    _title.setFillColor(sf::Color::Black);
+    _title.setString("Inventory");
+    _title.setPosition(_body.getPosition().x + 40, _body.getPosition().y + 10);
 }
 
 void Inventory::display()
 {
     _window->draw(_body);
     _window->draw(_button);
+    if (!_isOpen)
+        return;
+    _window->draw(_title);
+    _window->draw(_textPos);
+    for (auto &t : _textsInv)
+        _window->draw(t);
+    for (auto &t : _textsPlayer)
+        _window->draw(t);
+    for (auto &t : _textsEgg)
+        _window->draw(t);
+}
+
+void Inventory::initTextTilePosition()
+{
+    _textPos.setFont(_font);
+    _textPos.setCharacterSize(20);
+    _textPos.setFillColor(sf::Color::Black);
+    _textPos.setString(std::string("x: " + std::to_string(int(_tilePosition.x)) + "\n\n\ny: " + std::to_string(int(_tilePosition.y))));
+    _textPos.setPosition(_body.getPosition().x + 50, _body.getPosition().y + 60);
+}
+
+void Inventory::initTextTileInventory()
+{
+    sf::Text text;
+
+    for (auto &val : _tileInventory) {
+        text.setString(std::to_string(val));
+        _textsInv.push_back(text);
+    }
+    for (auto &t : _textsInv) {
+        t.setFont(_font);
+        t.setFillColor(sf::Color::Black);
+        t.setCharacterSize(20);
+    }
+    setPosTextsInv();
+}
+
+void Inventory::initTextPlayer()
```

Comment:
Could be generic because the next function is exactly the same, just add str entitiyType params to the function